src/stan/model/indexing/rvalue.hpp

```diff
@@ -731,13 +722,15 @@ inline auto rvalue(
   if (idxs.tail_.head_.is_ascending()) {
     const auto col_start = idxs.tail_.head_.min_ - 1;
     return rvalue(x.middleCols(col_start, idxs.tail_.head_.max_ - col_start),
-                  index_list(idxs.head_), name, depth + 1);
+                  index_list(idxs.head_), name, depth + 1)
+        .eval();
```

Comment:
Expressions not work here? Or are they of different types so they have to be eval'd?

Reply:
Yep same thing here with min_max